### PR TITLE
Add option for external auth server

### PR DESCRIPTION
**Pull request by cherbst**
This option causes the webapp to redirect to the given URL on page load
and also to not store any token in local storage

I release these changes under Apache License 2.0 and additionally authorize rami.io GmbH to distribute these changes under the same license as the rest of the project

### CODE REVIEW

File: server/venueless/core/models/world.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ class World(VersionedModel):
     )
     timezone = models.CharField(max_length=120, default="Europe/Berlin")
     feature_flags = JSONField(blank=True, default=default_feature_flags)
+    external_auth_url = models.CharField(max_length=255, default="")
```
Comment (raphaelm) [inline server/venueless/core/models/world.py:146]: I think ``models.URLField`` would be a better choice. Also, the default should not be an empty string, but a real null-value, e.g.

```
models.URLField(null=True, blank=True)
```

File: webapp/src/main.js [hunk 2]
```diff
@@ -43,9 +44,14 @@ async function init () {
...
 	const token = new URLSearchParams(router.currentRoute.hash.substr(1)).get('token')
 	if (token) {
-		localStorage.token = token
+		if (!config.externalAuthUrl) {
+			localStorage.token = token
+		}
 		router.replace(router.currentRoute.path)
 		store.dispatch('login', {token})
+	} else if (config.externalAuthUrl) {
+		location = config.externalAuthUrl
```
Comment (raphaelm) [inline webapp/src/main.js:53]: My personal taste would be to prefer `window.location` to make clear this is not a local variable of some kind

**Review by raphaelm — COMMENTED** [review]
Hi,

thank you very much for your contribution!

I added two minor style comments, and I also think the CI wants you to run `black .` on the server code to format it according to the project layout.

@rashfael Could you have a quick look at the frontend code as well if the logic is in (all) the right places, before we move forward with this?

@cherbst To be quite honest, this is the first code contribution we ever had for venueless from someone who is not on our payroll in some way. So we haven't really got a process for the license part yet. I don't think we need to make it complicated for small changes like this, but maybe you can put an explicit "I release these changes under Apache License 2.0 and additionally authorize rami.io GmbH to distribute these changes under the same license as the rest of the project" in your PR so we have at least a minimal paper trail for the changes?

**Comment by rashfael** [comment]
The new config attribute `externalAuthUrl` needs to be added to the docs https://github.com/venueless/venueless/blame/dev/docs/admin/webapp-config.rst/#L16.
It might also be better to put the redirect call before vue gets loaded to get the redirect out as fast as possible.
(Perhaps even putting token handling/saving outside of init and passing a token to `init`).

**Review by rashfael — APPROVED** [review]

**Pull request merged**